app/src/main/admin/components/Users/SingleUser/UserDetailsContent.tsx

```diff
@@ -0,0 +1,62 @@
+import * as React from "react";
+import {RemoteContent} from "../../../../shared/models/RemoteContent";
+import {RemoteContentComponent} from "../../../../shared/components/RemoteContentComponent/RemoteContentComponent";
+import {connectToStores} from "../../../../shared/alt";
+import {User} from "../../../../shared/models/Generated";
+import {UserRole} from "./UserRoleComponent";
+import {userStore} from "../../../stores/UserStore";
+interface Props extends RemoteContent {
+    user: User;
+}
+
+const commonStyles = require("../../../../shared/styles/common.css");
+
+export class UserDetailsContentComponent extends RemoteContentComponent<Props> {
+    static getStores() {
+        return [userStore];
+    }
+
+    static getPropsFromStores(): Props {
+        const user = userStore.getCurrentUserDetails();
+        return {
+            user: user,
+            ready: user != null
+        };
+    }
+
+    renderContent(props: Props) {
+
+        let roles;
+        if (props.user.roles != null)
+            roles =
+                <tr>
+                    <td>Roles</td>
+                    <td>
+                        {props.user.roles
+                            .map(r =>
+                                <UserRole key={ r.name + r.scope_prefix + r.scope_id } {...r} />)}
+                    </td>
+                </tr>;
```

Comment:
I'm generally not hopping keen on single line `ifs` with omitted braces, but I haven't been jumping on them in your code, because we don't have to agree on every code style point. But I really don't think this is clear - much better to have braces around this `if` statement.

Also, maybe separate out the role rendering logic into another method?